Automattic/wp-calypso
# Free trials: Start AB test

Fixes #1520 by offering free trials during signup, and thereafter on the /plans page, to 10% of users who are assigned the "offered" variation in the abtest.

#### Testing instructions (existing user)

1. Run `git checkout update/create-ab-test-for-free-trials` and start your server
2. Open the http://calypso.dev:3000/plans with an existing account
3. Assert that you are not offered a free trial

#### Testing instructions (new user, not in the test)

1. Run `git checkout update/create-ab-test-for-free-trials` and start your server
2. Open the http://calypso.dev:3000/start
3. Add yourself to the notOffered bucket for the abtest:
`localStorage.setItem('ABTests','{"freeTrials_20160107":"notOffered"}')`
3. Assert that you see the standard signup flow.

#### Testing instructions (new user, in the test)

1. Run `git checkout update/create-ab-test-for-free-trials` and start your server
2. Open the http://calypso.dev:3000/start
3. Add yourself to the notOffered bucket for the abtest:
`localStorage.setItem('ABTests','{"freeTrials_20160107":"offered"}')`
3. Assert that you see the free trial signup flow (/start/free-trial).
4. Assert that you are offered a free trial on the plans step
5. Assert that you are sent to /checkout with a trial in the cart after signup
6. Don't purchase the trial, and navigate to /plans
7. Assert that you are offered a free trial.

#### Additional notes

We should update the date before deploying this.

#### Reviews
